### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by glowcap** [baruma/SassyCloud#1]
Although the code is clean, it looks very much like a tutorial that was followed.

Some things to point out and change:
 - In Swift 4 there's no need for Swifty JSON. You should remove the Swifty JSON pod and change the parsing to use the default `Decodable` functionality of Swift
- There shouldn't be any forced unwrapped optionals (ex: `var opt1: String!`) in production code. These should be unwrapped with `guard` or `if let` depending on the situation.
- This is not MVC. Your networking should not be in the ViewController. The ViewController should only be for how the code should be displayed. Instead use protocols to include API calls.
- Remove any unnecessary comments from your code (these add to file size) before pushing to a remote

Sorry to sound very critical, but these are important things that will make or break a real interview assignment.

**Comment by baruma** [comment]
- Although the code is clean, it looks very much like a tutorial that was followed.

So I will say that I did not follow a tutorial app.  At most, I referred to an old tutorial for direction purposes early on for the JSON parsing and networking but that was the extent of it.  The majority of this was original code.  I would like to hear more on how to not make this seem as though it was followed by a tutorial though.  

- In Swift 4 there's no need for Swifty JSON. You should remove the Swifty JSON pod and change the parsing to use the default Decodable functionality of Swift

I originally didn't know about Codable provided by Swift and did default to SwiftyJSON but I did refactor that code to use Codable instead.  

- There shouldn't be any forced unwrapped optionals (ex: var opt1: String!) in production code. These should be unwrapped with guard or if let depending on the situation.
- This is not MVC. Your networking should not be in the ViewController. The ViewController should only be for how the code should be displayed. Instead use protocols to include API calls.
- Remove any unnecessary comments from your code (these add to file size) before pushing to a remote.
- When creating a Github project, you should also include a ReadMe file providing information about your project.

Noted.  This isn't the final version of the app (Just a working MVP) and I do plan on improving upon it by refactoring a lot of the code.  I will be adding a ReadMe later on again, as the app is further worked on beyond the MVP.

- Sorry to sound very critical, but these are important things that will make or break a real interview assignment.

I don't mind, and would rather prefer it since I can work on my faults and improve upon the project overrall.  Thank you for taking the time to leave me some feedback.

**Comment by glowcap** [comment]
> So I will say that I did not follow a tutorial app. 

Yep. That's why I said "look...like". The code looks like different code put together. It's a red flag for interviewers. The code should be consistent. Using two different parsing techniques in the same codebase could lose you a potential job.

> This isn't the final version of the app

I get that. I just critique it as if it was a final product because that's how interviewers will look at it. I also what to give you a direction and sense of what is expected of your code.

Keep in mind, a readme file is a priority for interviewers. Even though it's not part of the code, it reflects your ability to communicate. Many interviewers won't take the time to search through your code to understand what it's supposed to do.

**Comment by baruma** [comment]
- Yep. That's why I said "look...like". The code looks like different code put together. It's a red flag for interviewers. The code should be consistent. Using two different parsing techniques in the same codebase could lose you a potential job.

I gotcha.  Alright, I'll look into this issue further and see what I can do to work on this.

- I get that. I just critique it as if it was a final product because that's how interviewers will look at it. I also what to give you a direction and sense of what is expected of your code.

Got it!  Yeah, these are definitely some good points to continue to focus on then.  

- Keep in mind, a readme file is a priority for interviewers. Even though it's not part of the code, it reflects your ability to communicate. Many interviewers won't take the time to search through your code to understand what it's supposed to do.

I didn't realize this - thanks for pointing out the significance of the readme then.  Will definitely be sure to add that in the next update!
